### PR TITLE
Endpoints

### LINKED ISSUE #16: Routes
Create initial layout of routes for backend api calls

### PR DESCRIPTION
**Pull request by amsmyth1**
Create the API Contract Document

Routes and JSON Responses documented. 

close #16

### CODE REVIEW

File: APIContract.md [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+      "description": "This yard is cement",
+      "payment": "venmo/paypal infomraiton?",
+      "availability": "mornings",
+      "photo_url_1": "https:picture1.org",
+      "photo_url_2": "",
+      "photo_url_3": ""
+      }
+    }
+  }
+  ```
```
Comment (abreaux26) [inline APIContract.md]: Should there just be one closing curly bracket at the end? (Same question for the `Update a yard` and `Create a Booking`)

File: APIContract.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+        }
+    }
+  ]
+}
+```
+
+## Booking Details
+`GET /api/v1/bookings/:booking_id`
+  ```json
+    "data": {
```
Comment (abreaux26) [inline APIContract.md]: I think you are missing a starting curly brace before data?

**Comment by abreaux26** [comment]
Nice work!! This looks amazing!! and will help us tremendously moving forward!

File: APIContract.md (hunk 1 above)
Comment (amsmyth1) [reply APIContract.md]: Great catch! I corrected the syntax on these endpoint responses

File: APIContract.md (hunk 2 above)
Comment (amsmyth1) [reply APIContract.md]: Curly bracket added. Thanks for the thorough review!

**Comment by amsmyth1** [comment]
Hey Angel - thanks for the thorough review. All corrections have been made.

**Review by abreaux26 — APPROVED** [review]

File: APIContract.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
+# Yardsourcing API Contract
+
+## Endpoints
+
+| HTTP verbs | Paths  | Used for | Output |
+| ---------- | ------ | -------- | ------:|
```
Comment (Gvieve) [inline APIContract.md:6]: I think we may need to add a few endpoints, let me know what you ladies think.
__purposes__
- GET to get a list of available purposes we can present on Create yard form as checkboxes

__yard_purposes__
- POST create a yard_purpose record(s) when a yard is created, or new purpose is added
- DELETE, destroy a yard purpose record when the yard still exists but the host wishes to remove an existing purpose

File: APIContract.md [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+        "price": 10.50,
+        "description": "This yard is green and 1200 sf",
+        "payment": "venmo/paypal infomraiton?",
+        "availability": "weekends and weekdays after 5pm",
+        "photo_url_1": "https:picture1.org",
+        "photo_url_2": "",
+        "photo_url_3": ""
+      }
+    }
+  }
```
Comment (Gvieve) [inline APIContract.md:97]: I think all of the yard information responses back may need to include purpose data as another nested collection.

File: APIContract.md [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+    "date": "08-15-21",
+    "time": "5:00PM",
+    "duration": 120,
+    "description": "Renting a yard to run my pup and their best friend for 2 hours"
+  }
+  ```
+
+## Delete a Booking
+`DELETE /api/v1/bookings/:booking_id`
+- Destroy a booking if the id matches and it passes cancellation policy validation
```
Comment (Gvieve) [inline APIContract.md:564]: Where does the cancellation policy currently live?

File: APIContract.md [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+| GET | /api/v1/renters/renter_id/bookings?status=STATUS | Get all bookings for a renter that have a particular status| [json](#get-user-bookings-by-status) |
+| GET | /api/v1/yards?location=ZIP&purposes=PURPOSE+NAME&OTHER+PURPOSE+NAME  | Get yards that match search criteria. | [json](#yard-search) |
+| GET | /api/v1/hosts/host_id/yards  | Get yards that belong to a host | [json](#user-yards) |
+| GET | /api/v1/yards/:yard_id/bookings | Get bookings that belong to a specific yard | [json](#yard-bookings) |
+| POST | /api/v1/bookings | Create a new booking | [json](#create-a-booking) |
+| GET | /api/v1/bookings/:booking_id | Get booking show page | [json](#booking-details) |
+| DELETE | /api/v1/bookings/:booking_id | Delete a new booking | [details](#delete-a-booking) |
+
+
+## JSON Responses
```
Comment (Gvieve) [inline APIContract.md:23]: How are we doing error handling?

File: APIContract.md (hunk 3 above)
Comment (jbranham1) [reply APIContract.md:6]: Yes!! Great idea

File: APIContract.md (hunk 6 above)
Comment (jbranham1) [reply APIContract.md:23]: @amsmyth1 We should brainstorm and write out some error responses!

File: APIContract.md (hunk 3 above)
Comment (amsmyth1) [reply APIContract.md:6]: Great catch.  I will add these now.

File: APIContract.md (hunk 4 above)
Comment (amsmyth1) [reply APIContract.md:97]: Added per our meeting

File: APIContract.md (hunk 5 above)
Comment (amsmyth1) [reply APIContract.md:564]: Per meeting the cancellation policy will live in the front end (potentially as an enviro variable) that disables a cancel button if the booking is with the policy.

**Comment by amsmyth1** [comment]
All changes have been made per group meeting 4/14/2021 at 3pm.

**Review by domo2192 — APPROVED** [review]

**Pull request merged**